crates/forge/src/lib.rs

```diff
@@ -216,21 +209,22 @@ fn to_runnable(
 /// * `runner_config` - A configuration of the test runner
 /// * `corelib_path` - Absolute path to the Cairo corelib
 /// * `contracts` - Map with names of contract used in tests and corresponding sierra and casm artifacts
-/// * `predeployed_contracts` - Absolute path to predeployed contracts used by starknet state e.g. account contracts
 ///
 
 #[allow(clippy::implicit_hasher)]
 pub async fn run(
-    package_path: &Utf8Path,
     package_name: &str,
-    package_source_dir_path: &Utf8Path,
+    snforge_target_dir_path: &Utf8Path,
     tests_filter: &TestsFilter,
     runner_config: Arc<RunnerConfig>,
     runner_params: Arc<RunnerParams>,
 ) -> Result<Vec<TestCrateSummary>> {
-    let compilation_targets =
-        collect_test_compilation_targets(package_path, package_name, package_source_dir_path)?;
-    let test_crates = compile_tests(&compilation_targets, &runner_params)?;
+    let snforge_test_artifact_path =
+        snforge_target_dir_path.join(format!("{package_name}.sierra.json"));
+    let test_crates = serde_json::from_str::<Vec<CompiledTestCrateRaw>>(&std::fs::read_to_string(
+        snforge_test_artifact_path,
+    )?)?;
```

Comment:
If the artefacts compilation is going on in scarb now, perhaps this function should already accept them in the deserialized (as structs/vec/whatever rust object) instead of taking the path. 

This way we can still use it in integration tests in an easier way.

Reply:
This would put even more logic into `main.rs`, I don't really like that. Also it doesn't simplify integration tests imo, we don't want to generate them manually for tests anyways

Reply:
This doesn't necessarily have to be in main, it can be a separate function in lib that would be called before `run`. Similarly to how collecting and compilation was split into multiple functions called in a sequence

Reply:
I think we just want to replace the collection + compilation logic here, no? So it would make sense to read it here I think, I can extract it to a separate function that with a docstring explaining what it does

Reply:
This seems like a good middle ground for now, we can do that.